test/unit/collection-view.spec.js

```diff
@@ -532,6 +585,40 @@ describe('collection view', function() {
     });
   });
 
+  describe('when a model that does not pass the filter is removed from the collection', function() {
+    beforeEach(function() {
+      this.model = new Backbone.Model({foo: 'bar'});
+      this.rejectedModel = new Backbone.Model({ foo: 'rejected' });
+      this.collection = new Backbone.Collection([this.model, this.rejectedModel]);
+
+      this.CollectionView = this.MockCollectionView.extend({
+        indexToReject: 1,
+        filter: function(model, index) {
+          return index !== this.indexToReject;
```

Comment:
This one specifically was adding a test for filtering by index. I could move it to a separate block, though.